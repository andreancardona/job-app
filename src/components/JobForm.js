import React, { Fragment }from 'react';
import '../styles/_job-form.scss';

const JobForm = (props) => {
  const handleInputChange = (event) => {
		const { name, value } = event.target

		props.setJob({ ...props.job, [name]: value })
	}

  // Update header and subHeader based current form
  const header = props.adding ? 'Add a new job' : 'Edit a job';

  const subHeader = props.adding ? 'Fill out the information for your new job listing.' : 'Edit the information for your job listing.';

  return (
    <Fragment>
      <div className="form-headers" tabIndex="0" role="heading" aria-level="1" aria-label={header + subHeader}>
        <div className="form-heading">
          {header}
        </div>
        <div className="form-sub-heading">
          {subHeader}
        </div>
      </div>
      <form className="job-form">
        <label className="label" tabIndex="0">{props.labels.title} *</label>
        <div className="sub-label" tabIndex="0">{props.subLabels.title}</div>
          <input className="input" tabIndex="0" type="text" name="title" value={props.job.title} onChange={handleInputChange} />
        <label className="label" tabIndex="0">{props.labels.location} *</label>
        <div className="sub-label" tabIndex="0">{props.subLabels.location}</div>
          <input className="input" type="text" name="location" value={props.job.location}tabIndex="0"onChange={handleInputChange} />
        <label className="label" tabIndex="0">{props.labels.sponsorship} *</label>
        <div className="sub-label" tabIndex="0">{props.subLabels.sponsorship}</div>    
          <select className="select-input" name="sponsorship" onChange={handleInputChange} value={props.job.sponsorship}>
            <option value="Sponsor">Sponsor</option>
            <option value="Free">Free</option>
          </select>
        <label className="label" tabIndex="0">{props.labels.status} *</label>
        <div className="sub-label" tabIndex="0">{props.subLabels.status}</div> 
          <select className="select-input" onChange={handleInputChange} name="status" value={props.job.status} tabIndex="0">
            <option value="Open">Open</option>
            <option value="Paused">Paused</option>
            <option value="Closed">Closed</option>
          </select>
      </form>
    </Fragment>  
  )
}

export default JobForm;