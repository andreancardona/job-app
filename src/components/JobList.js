import React, { useContext } from "react";
import { Link } from "react-router-dom";
import { JobContext } from '../context/JobContext';
import CurrentDate from '../utils/CurrentDate';

const JobList = () => {

  const { jobs } = useContext(JobContext);

  const job = jobs.map(job => {
    return job;
  })

  const jobListHeaders = {
    title: 'Job Title',
    posted: 'Posted',
    sponsorship: 'Sponsorship',
    status: 'Status'
  }

  return (
    <div className="jobs-container">
      <div className="header-container">
        <div className="header-jobs" tabIndex="0" role="heading" aria-level="1">
          Jobs 
          <div className="listings-count">{jobs.length} listings</div>
        </div>
        <Link to={{pathname: `/new-job-details/${job.id}`}}>
         <button className="add-save-button">ADD JOB</button>
       </Link>
      </div>
      <div aria-hidden="true" className="jobs-list-container" role="grid" id="job-list">
        <div id="job-list-headers" className="jobs-list-headers" tabIndex="0" role="row" aria-label="jobs listings">
          <div className="header-title" aria-label="title">{jobListHeaders.title}</div>
          <div className="header-posted" aria-label="posted">{jobListHeaders.posted}</div>
          <div className="header-sponsorship"  aria-label="sponsorhsip">{jobListHeaders.sponsorship}</div>
          <div className="header-status" aria-label="status">{jobListHeaders.status}</div>
        </div>
        {jobs.map(job => {
          return (
            <div className="job-card" tabIndex="0" key={job.id}>
              <div className="job-title-location">
                <div className="title" aria-label={"title" + job.title}>{job.title}</div>
                <div className="location" aria-label={"location" + job.location}>{job.location}</div>
              </div>
              <div className="job-post-details">
                <div className="posted" aria-label={"posted" + job.posted}>{job.posted ? job.posted : <CurrentDate />}</div>
                <div className="sponsorship" aria-label={"sponsorship" + job.sponsorship}>{job.sponsorship}</div>
                <div className="status" aria-label={"status" + job.status}>{job.status}</div>
                <Link to={{pathname: `/new-job-details/${job.id}`}}>
                  <button className="edit-button">Edit</button>
                </Link>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default JobList;