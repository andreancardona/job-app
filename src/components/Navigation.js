import React from 'react';
import profilePlaceholder from '../images/profilePlaceholder.svg';
import '../styles/Navigation.css';

const Navigation = () => {
  return (
    <div className="nav-container">
      <div className="nav-title" tabIndex="0" >Welcome to JobHob</div>
        <div className="user-profile">
          <img className="user-profile--image" alt="profile placeholder" tabIndex="0"  src={profilePlaceholder} />
          <div className="user-profile--info">
            <div className="user-profile--name" tabIndex="0">Andrea Cardona</div>
            <div className="user-profile--role" tabIndex="0">Admin</div>
          </div>
        </div>
     </div>
  );
}

export default Navigation;