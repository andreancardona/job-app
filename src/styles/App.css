 /* banner */

.banner {
  position: absolute;
  height: 64px;
  left: 0px;
  top: 0px;
  background: #2D2D2D;
  width: 100%;
}