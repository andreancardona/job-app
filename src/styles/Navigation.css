/* banner */

.banner {
  position: absolute;
  height: 64px;
  left: 0px;
  top: 0px;
  background: #2D2D2D;
  width: 100%;
}
 
 /* navigation */

 .nav-container {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  width: 100%;
 }

.header {
  color: white;
  font-family: Helvetica Neue;
  margin: 20px 20px 0 20px;
  font-size: 16px;
  font-style: normal;
  font-weight: 700;
  letter-spacing: 0em;
  height: 10px;
}

/* user profile */

.user-profile--image {
  width: 36px;
  height: 36px;
  left: 822px;
  margin: 12px 8px 0 0;
}

.user-info {
  display: flex;
  flex-direction: column;
  margin-top: 11px;
}

.user-profile--name {
  left: 866px;
  font-family: Helvetica Neue;
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  line-height: 150%;
  color: #FAF9F8;
}

.user-profile--role {
  width: 35px;
  height: 18px;
  left: 866px;
  top: 35px;
  font-family: Helvetica Neue;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 150%;
  color: #FAF9F8;
}