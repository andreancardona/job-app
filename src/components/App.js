import React, { useState } from 'react';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Navigation from './Navigation';
import Header from './Header';
import JobsList from '../components/JobsList';
import JobForm from './JobForm';

const App = () => {

  // Navigation props
  const navProps = {
    title: 'JobHob',
    userName: 'Andrea Cardona',
    userRole: 'Admin'
  }

  // Initial list of jobs
  const jobsList = [
    { id: 1, title: 'Product Manager', location: 'Austin, TX', posted: '10/04/2020', sponsorship: 'Free', status: 'Open' },
    { id: 2, title: 'CEO', location: 'Austin, TX', posted: '12/30/2020', sponsorship: 'Sponsored', status: 'Paused'},
    { id: 3, title: 'Software Engineer', location: 'Seattle, WA', posted: '11/18/2020', sponsorship: 'Free', status: 'Closed' },
  ]

  // Initial state of our job form
  const initialJobForm = { id: '', title: '', location: '', sponsorship: '', status: '' };

  // Set our state / read our list of jobs
  const [ jobs, setJobs ] = useState(jobsList)
  const [ job, setJob ] = useState(initialJobForm);
	const [ currentJob, setCurrentJob ] = useState(initialJobForm)
	const [ editing, setEditing ] = useState(false)
  const [ adding, setAdding ] = useState(false)

  // Update set add job state
  const setAdd = () => {
    setAdding(true);
  }

  // Add a job
  const addJob = (job) => {
		job.id = jobs.length + 1
    const newJobs = [...jobs, job]

		setJobs(newJobs);
	}

  // Allow user to edit job
  const editJob = (job) => {
    setEditing(true)

    setCurrentJob({ 
      id: job.id, 
      title: job.title, 
      location: job.location, 
      sponsorship: job.sponsorship, 
      status: job.status 
    })
  }

  // Update a job
  const updateJob = (id, jobBeingUpdated) => {
		setEditing(false)
    const updatedJob = jobs.map(job => (job.id === id ? jobBeingUpdated : job))

		setJobs(updatedJob);
	}

  return (
    <div role="contentinfo" className="app-container">
      <Navigation props={navProps}/>
      <Router>
          <Switch>
            <Route path="/home" component={() => 
              <main id="main" role="main">
                <Header jobs={jobs} setAdd={setAdd}/>
                <JobsList jobs={jobs} editJob={editJob}/>
              </main>
            } />
            <Route path="/job-form" component={() => 
              editing
              ? <JobForm
                  job={job}
                  jobs={jobs} 
                  editing={editing}
                  adding={adding}
                  setJob={setJob}
                  addJob={addJob} 
                  editJob={editJob}
                  currentJob={currentJob}
                  updateJob={updateJob}
                  setAdding={setAdding} 
                  setEditing={setEditing}
                /> 
              : 
                <JobForm
                  job={job}
                  jobs={jobs} 
                  editing={editing}
                  adding={adding}
                  setJob={setJob}
                  addJob={addJob} 
                  editJob={editJob}
                  currentJob={currentJob}
                  updateJob={updateJob}
                  setAdding={setAdding} 
                  setEditing={setEditing}
                />}
             /> 
          </Switch>
      </Router>
    </div>
  )
};

export default App;
