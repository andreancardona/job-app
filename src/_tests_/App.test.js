import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import App from '../components/App';
import Navigation from '../components/Navigation';
import JobList from '../components/JobList';
import JobForm from '../components/JobForm';
import JobProvider from '../context/JobContext';

render(<JobContext.Provider value={'a value that makes sense'}></JobContext.Provider>)

describe('The <App /> component', () => {
  test('App renders without crashing - hopefully', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  test('App should not fail any accessibility tests', async () => {
    const { container } = render(<App />);
    expect(await axe(container)).toHaveNoViolations();
  });

  test('Navigation renders', () => {
    const tree = renderer.create(<Navigation />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('JobList renders', () => {
    const tree = renderer.create(<JobList />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('JobsForm renders with JobsForm props', () => {
    const tree = renderer.create(<JobForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});