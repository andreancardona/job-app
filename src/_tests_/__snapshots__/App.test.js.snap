// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <App /> component Header renders with header props 1`] = `
<div
  className="header-container"
>
  <div
    aria-level="1"
    className="header-jobs"
    role="heading"
    tabIndex="0"
  >
    Jobs
    <div
      className="listings-count"
    >
      3
       listings
    </div>
  </div>
  <button
    className="add-save-button"
    onClick={[Function]}
  >
    Add job
  </button>
</div>
`;

exports[`The <App /> component JobsList renders with JobsList props 1`] = `
<div
  aria-hidden="true"
  className="jobs-list-container"
  id="job-list"
  role="grid"
>
  <div
    aria-label="jobs listings"
    className="jobs-list-headers"
    id="job-list-headers"
    role="row"
    tabIndex="0"
  >
    <div
      aria-label="title"
      className="header-title"
    >
      Job Title
    </div>
    <div
      aria-label="posted"
      className="header-posted"
    >
      Posted
    </div>
    <div
      aria-label="sponsorhsip"
      className="header-sponsorship"
    >
      Sponsorship
    </div>
    <div
      aria-label="status"
      className="header-status"
    >
      Status
    </div>
  </div>
  <div
    className="job-card"
    role="row"
    tabIndex="0"
  >
    <div
      className="job-title-location"
    >
      <div
        aria-label="titleProduct Manager"
        className="title"
      >
        Product Manager
      </div>
      <div
        aria-label="locationAustin, TX"
        className="location"
      >
        Austin, TX
      </div>
    </div>
    <div
      className="job-post-details"
    >
      <div
        aria-label="posted10/04/2020"
        className="posted"
      >
        10/04/2020
      </div>
      <div
        aria-label="sponsorshipFree"
        className="sponsorship"
      >
        Free
      </div>
      <div
        aria-label="statusOpen"
        className="status"
      >
        Open
      </div>
      <button
        aria-label="edit job"
        className="edit-button"
        onClick={[Function]}
        tabIndex="0"
      >
        Edit
      </button>
    </div>
  </div>
  <div
    className="job-card"
    role="row"
    tabIndex="0"
  >
    <div
      className="job-title-location"
    >
      <div
        aria-label="titleCEO"
        className="title"
      >
        CEO
      </div>
      <div
        aria-label="locationAustin, TX"
        className="location"
      >
        Austin, TX
      </div>
    </div>
    <div
      className="job-post-details"
    >
      <div
        aria-label="posted12/30/2020"
        className="posted"
      >
        12/30/2020
      </div>
      <div
        aria-label="sponsorshipSponsored"
        className="sponsorship"
      >
        Sponsored
      </div>
      <div
        aria-label="statusPaused"
        className="status"
      >
        Paused
      </div>
      <button
        aria-label="edit job"
        className="edit-button"
        onClick={[Function]}
        tabIndex="0"
      >
        Edit
      </button>
    </div>
  </div>
  <div
    className="job-card"
    role="row"
    tabIndex="0"
  >
    <div
      className="job-title-location"
    >
      <div
        aria-label="titleSoftware Engineer"
        className="title"
      >
        Software Engineer
      </div>
      <div
        aria-label="locationSeattle, WA"
        className="location"
      >
        Seattle, WA
      </div>
    </div>
    <div
      className="job-post-details"
    >
      <div
        aria-label="posted11/18/2020"
        className="posted"
      >
        11/18/2020
      </div>
      <div
        aria-label="sponsorshipFree"
        className="sponsorship"
      >
        Free
      </div>
      <div
        aria-label="statusClosed"
        className="status"
      >
        Closed
      </div>
      <button
        aria-label="edit job"
        className="edit-button"
        onClick={[Function]}
        tabIndex="0"
      >
        Edit
      </button>
    </div>
  </div>
</div>
`;

exports[`The <App /> component Navigation renders with nav props 1`] = `
<nav
  className="nav-container"
  id="nav"
>
  <div
    className="nav-title"
  >
    JobHob
  </div>
  <a
    className="skip-content"
    href="#main"
  >
    Skip to main content
  </a>
  <div
    className="user-profile"
  >
    <img
      alt="profile placeholder"
      className="user-profile--image"
      src="profilePlaceholder.svg"
    />
    <div
      className="user-profile--info"
    >
      <div
        className="user-profile--name"
      >
        Andrea Batlle
      </div>
      <div
        className="user-profile--role"
      >
        Admin
      </div>
    </div>
  </div>
</nav>
`;
